### DISCUSSION TITLE
Accessing Vector Clusters

### DISCUSSION THREAD

**Discussion by dee-corr** [milvus-io/milvus#19614] [Q&A and General discussion]
Hello,

I understand that Milvus uses the chosen index type and similarity metric to cluster the vectors. It would be of great benefit to gain access to these clusters and visualise them graphically. Is this possible? What I'm looking for is basically an id for each cluster, metadata e.g. what its center point is, and the list of vector ids within it.

**Comment by yhmo** [comment]
This tool might can help: https://github.com/zilliztech/feder
It is a visualization tool for hnsw, faiss index(IVF series). I am not sure it can display vector ids. You can contact the maintainers([shanghaikid](https://github.com/zilliztech/feder/commits?author=shanghaikid), [alwayslove2013](https://github.com/zilliztech/feder/commits?author=alwayslove2013)) for more information.

**Reply by dee-corr** [reply]
Thank you @yhmo ! This looks amazing.

On the milvus side, there any resource which explains how to get the lists of vectors in each cluster?

**Reply by yhmo** [reply]
If you want to get all vector ids, the query() is the only one choice.
```
    collection.load()
    result = collection.query(expr="id_field > 0", output_fields=["id_field"])
    for res in result:
        print(res)
```
